src/lib/ares_process.c

```diff
@@ -71,35 +71,66 @@ static void          end_query(ares_channel_t *channel, struct query *query,
                                ares_status_t status,
                                const ares_dns_record_t *dnsrec);
 
-static void          server_increment_failures(struct server_state *server)
+static void server_increment_failures(struct server_state *server)
 {
-  ares__slist_node_t   *node;
-  const ares_channel_t *channel = server->channel;
+  ares__slist_node_t *node;
+  ares_channel_t     *channel = server->channel;
+  char                server_ip[INET6_ADDRSTRLEN];
 
   node = ares__slist_node_find(channel->servers, server);
   if (node == NULL) {
     return;
   }
+
+  server->consec_successes = 0;
   server->consec_failures++;
   ares__slist_node_reinsert(node);
+
+  /* If the server has just become unhealthy, then invoke the server state
+   * callback.
+   */
+  if ((server->consec_failures >= channel->server_failure_threshold) &&
+      (server->is_healthy == ARES_TRUE)) {
```

Comment:
Avoid extra parens unless it really adds clarity due to order of operations.

Reply:
Thanks, I will get these all updated.